main.cpp

```diff
@@ -99,6 +93,24 @@ float CentralProcessingUnit::processData()
     return processedData;
 }
 
+struct cpuCounter
+{
+    CentralProcessingUnit cpu1;
+    int startCounter(int initVal, int maxVal)
+    {
+        int i = 0;
+        cpu1.counter = initVal;
+        while (i < maxVal)
```

Comment:
you never increment `i`, so the only thing making this loop exit is the check on line 106.
- [ ] revise.  increment `i` or change your `while( expr )`